node/network/approval-distribution/src/lib.rs

```diff
@@ -841,6 +851,8 @@ impl State {
 		peer_id: PeerId,
 		view: View,
 	) {
+		const MAX_DEPTH_TO_UNIFY: usize = 4;
```

Comment:
With this approach, what should happen if there's a block from long ago which is unapproved, and there is for some reason churn in the peer-set? It seems to me that the assignment/approvals would not propagate and that would cause a finality stall.

Reply:
Good point. An alternative to this would be adding a min height peer is interested in, but as it's a user controlled setting, it doesn't solve the problem of asking for a lot of assignments. And currently we sometimes send/receive A LOT of approvals/assignments
> Processing approvals from a peer peer_id=12D... num=708

Wouldn't we have the same problem if all validators are restarted as approvals dbs are ephemeral?

Reply:
> An alternative to this would be adding a min height peer is interested in, but as it's a user controlled setting, it doesn't solve the problem of asking for a lot of assignments

Yeah, it doesn't solve that. Maybe we should always rebroadcast messages which originated with local source, but not others.

> Wouldn't we have the same problem if all validators are restarted as approvals dbs are ephemeral?

Yes, but only if all validators are restarted. That is something we can address afterwards by making the DB less ephemeral.

Reply:
> Yeah, it doesn't solve that. Maybe we should always rebroadcast messages which originated with local source, but not others.

what about a hybrid approach when we propagate only local messages after a certain depth
bd56528401d2f578ce35a55762cf8d293785a5be